traits/testing/tests/test_unittest_tools.py

```diff
@@ -28,6 +26,11 @@
     on_trait_change,
 )
 from traits.testing.api import UnittestTools
+# unittest_tools provides a reference to unittest for historical reasons, and
```

Comment:
do we want to deprecate any such imports, similar to what we were doing with other adaptation-related imports?

Reply:
Ideally yes, but how would you propose implementing such a deprecation?

Reply:
I suppose one could add code somewhere at a traits top-level that adds an import hook that complains about `from traits.testing.unittest_tools import unittest` imports, but that seems a bit too magic for my tastes.

Reply:
Something I came across when I was purging `traits.protocols`

```
adapts = deprecated("use 'adapts' in 'traits.api' instead")(
    traits.adaptation.adapter.adapts
)
```

Reply:
Right, but that's for a `DeprecationWarning` that gets executed at function call time. How would you warn at _import_ time?

Reply:
Found the original usage - https://github.com/enthought/traits/blob/83df185600ab5662a1306326286540620b3fbe1c/traits/adapter.py 

Never ever came across anything like this before but it looked interesting. Dunno if it's the best solution to the problem though, assuming we even want to solve the problem.

Reply:
> Right, but that's for a DeprecationWarning that gets executed at function call time. How would you warn at import time?

Ohh. Yeah. You're right. Not sure how to get the behavior at import time.

Reply:
I mean, I think it could be done, but I can't see an easy way that isn't horribly magic. E.g., you could have `traits.testing.unittest_tools.unittest` be a wrapper around the real `unittest`, all of whose functions and classes issued a `DeprecationWarning` when you tried to use them. But I think the cure is worse than the disease.

Reply:
never mind then. do you want to update the comments to reflect when the API will be removed? Like with `traits.protocols`?

Reply:
(Of course, in Python 3.7, there's [PEP 562](https://www.python.org/dev/peps/pep-0562/))